### DISCUSSION TITLE
Security Fix: Prevent privilege escalation on secure screens

### DISCUSSION THREAD

**Discussion by seanbudd** [nvaccess/nvda#13379] [General]
Use this discussion for any technical questions surrounding the security fix for privilege escalation on secure screens. Please [responsibly disclose](https://en.wikipedia.org/wiki/Coordinated_vulnerability_disclosure) potential security issues to [info@nvaccess.org](mailto:info@nvaccess.org).

NVDA 2021.3.2 introduces a fix that prevents privilege escalation from secure screens. When performing an administrative action that requires a secure screen, a user can open a command prompt with system privileges. 

To enter a secure screen:

- Run an application as administrator, causing the User Access Control dialog to appear
- Enable NVDA to run during sign-in, then lock the desktop and continue to the password entry sign-in screen.

To reproduce using the addons manager dialog:

1. Enter a secure screen
2. Use an input gesture to open the addons manager dialog
3. Attempt to install an addon, which opens a file explorer browse dialog with system privileges
4. Open the context menu in a directory
5. Run command prompt with system privileges

To reproduce using context help:

1. Enter a secure screen
2. Press F1, to open context help in Internet explorer with system privileges
3. Press Control+O, to open a file explorer browse dialog with system privileges
4. Open the context menu in a directory
5. Run command prompt with system privileges

NVDA now disables context help and the addons manager dialog in secure screens. (#13059, #13353)

For further information, please contact NV Access via [info@nvaccess.org](mailto:info@nvaccess.org).

**Comment by Brian1Gaff** [comment]
I don't get the change to .3, when its the same as.2 though, Since the early 
downloads will have been .2 even though they did not say so, where is the 
harm in calling it .2?

Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "Sean Budd" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Tuesday, February 22, 2022 8:41 AM
Subject: [nvaccess/nvda] Security Fix: Prevent privilege escalation on 
secure screens (Discussion #13379)

**Reply by seanbudd** [reply]
I'm going to hide this as off-topic. 
As per [What's new in NVDA](https://www.nvaccess.org/files/nvda/releases/2021.3.3/documentation/changes.html)

>This release is identical to 2021.3.2. A bug existed in NVDA 2021.3.2 where it incorrectly identified itself as 2021.3.1. This release correctly identifies itself as 2021.3.3.

`buildVersion.version_minor` was incorrectly set to "1" in 2021.3.2.
